addons/l10n_sa_edi/models/account_journal.py

```diff
@@ -0,0 +1,645 @@
+                This will return three things:
+                    -   X509 Production Certificate 
+                    -   Password (Secret)
+                    -   Production Request ID
+    """
+
+    l10n_sa_csr = fields.Binary(attachment=True, copy=False, compute="_l10n_sa_compute_csr", store=True,
+                                groups="base.group_system",
+                                help="The Certificate Signing Request that is submitted to the Compliance API")
+    l10n_sa_csr_errors = fields.Html("Onboarding Errors")
+
+    l10n_sa_compliance_csid_json = fields.Char("CCSID JSON", copy=False, groups="base.group_system",
+                                               help="Compliance CSID data received from the Compliance CSID API "
+                                                    "in dumped json format")
+    l10n_sa_production_csid_json = fields.Char("PCSID JSON", copy=False, groups="base.group_system",
+                                               help="Production CSID data received from the Production CSID API "
+                                                    "in dumped json format")
+    l10n_sa_production_csid_validity = fields.Datetime("PCSID Expiration", help="Production CSID expiration date",
+                                                       compute="_l10n_sa_compute_production_csid_validity", store=True)
+    l10n_sa_compliance_checks_passed = fields.Boolean("Compliance Checks Done", default=False, copy=False,
+                                                      help="Specifies if the Compliance Checks have been completed successfully")
+
+    l10n_sa_chain_sequence_id = fields.Many2one('ir.sequence', string='ZATCA account.move chain sequence',
+                                                readonly=True, copy=False)
+
+    l10n_sa_serial_number = fields.Char("Serial Number", copy=False,
+                                        help="The serial number of the Taxpayer solution unit. Provided by ZATCA")
+
+    l10n_sa_latest_submission_hash = fields.Char("Latest Submission Hash", copy=False,
+                                                 help="Hash of the latest submitted invoice to be used as the Previous Invoice Hash (KSA-13)")
+
+    # ====== Utility Functions =======
+
+    def _l10n_sa_ready_to_submit_einvoices(self):
+        """
+            Helper function to know if the required CSIDs have been obtained, and the compliance checks have been
+            completed
+        """
+        self.ensure_one()
+        return self.sudo().l10n_sa_production_csid_json
+
+    # ====== CSR Generation =======
+
+    def _l10n_sa_csr_required_fields(self):
+        return ['l10n_sa_private_key', 'vat', 'name', 'city', 'country_id', 'state_id']
+
+    @api.depends('company_id.l10n_sa_private_key', 'l10n_sa_serial_number',
+                 'company_id.vat', 'company_id.name', 'company_id.city',
+                 'company_id.country_id', 'company_id.state_id')
```

Comment:
@mebe-odoo What is the idea still behind this computed field?  If someone changes something by accident to the company, he will have to re-onboard every journal.  Why not keep it as a normal field?

Reply:
@jco-odoo Done